src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -545,43 +545,39 @@ private boolean offerNotEnoughResources(Offer offer, double cpus, int mem) {
   }
 
   private void reconcileTasks(SchedulerDriver driver) {
-    updateReconciliationDeadline();
     driver.reconcileTasks(Collections.<Protos.TaskStatus> emptyList());
-    reconcilePersistentState();
-    reconciliationCompleted = true;
+    Timer timer = new Timer();
+    timer.schedule(new ReconcileStateTask(), conf.getReconciliationTimeout() * 1000);
```

Comment:
Reconcile only happens within this method so I'm not sure why the timer needs to be global. Everytime we reconcile we will call this method.